### DISCUSSION TITLE
visualization of latent layer and inducing points

### DISCUSSION THREAD

**Discussion by findoctorlin** [cornellius-gp/gpytorch#2251] [Q&A]
Hey everyone,

Is it possible that within our Gpytorch framework, that we visualize the inducing points and latent layer?

With the DGP regression example in the example folder (_https://github.com/cornelliusgp/gpytorch/tree/master/examples/05_Deep_Gaussian_Processes_) , which implement the idea of Prof. Deisenroth in the paper "Doubly Stochastic Variational Inference for Deep Gaussian Processes"(https://arxiv.org/abs/1705.08933) I build the model and applied it with a toy dataset and plot the result like this:
<img width="685" alt="image" src="https://user-images.githubusercontent.com/57867183/212348813-23b97060-e3ed-4723-934f-caf321b1466a.png">

Besides this, can we also visualize the mapping from input layer to the latent layer, then the mapping from latent layer to output layer, within our Gpytroch framework?

Thanks!

**Comment by michaelcao28** [comment]
hi @findoctorlin, have you been able to implement this? I'm trying to visualise a latent GP, where each latent GP has a different kernel structure. Any help would be appreciated!

**Comment by gpleiss** [comment]
Visualizing inducing points for DeepGP is likely going to be quite difficult. For non-deep GP, see also this issue: https://github.com/cornellius-gp/gpytorch/issues/1296#issuecomment-699509297
